src/components/Gallery/Gallery.js

```diff
@@ -0,0 +1,345 @@
+    ':hover': {
+      transition: 'opacity 0.5s linear',
+      opacity: '0.7'
+    }
+  }),
+  navButton: css({
+    color: '#fff',
+    display: 'none',
+    [lUp]: {
+      display: 'block',
+    }
+  }),
+}
+
+function removeQuery(url) {
+  return (url || '').split('?')[0]
+}
+
+class Gallery extends Component {
+  constructor(props) {
+    super(props)
+    const startItemIndex = props.items.findIndex(
+      i => removeQuery(i.src) === removeQuery(props.startItem)
+    )
+    this.state = {
+      index: startItemIndex > -1 ? startItemIndex : 0,
+      focus: false,
+    }
+
+    this.handleClickLeft = () => {
+      const total = this.props.items.length
+      this.setState(prevState => ({
+        index: prevState.index !== 0 ? (prevState.index - 1) : (total - 1),
+        exitLeft: false,
+        exitRight: false,
+      }))
+    }
+
+    this.handleClickRight = () => {
+      const total = this.props.items.length - 1
+      this.setState(prevState => ({
+        index: prevState.index !== total ? (prevState.index + 1) : 0,
+        exitLeft: false,
+        exitRight: false,
+      }))
+    }
+
+    this.toggleFocus = () => {
+      this.setState(prevState => ({ focus: !prevState.focus }))
+    }
+
+    this.handleSwipeDown = () =>
+      this.setState(
+        {closing: true},
+        debounce(props.onClose, swipeAnimationDurationMs)
+      )
+
+    this.handleSwipeLeft = () =>
+      this.setState(
+        {exitLeft: true},
+        debounce(this.handleClickRight, swipeAnimationDurationMs * 0.7)
+      )
+
+    this.handleSwipeRight = () =>
+      this.setState(
+        {exitRight: true},
+        debounce(this.handleClickLeft, swipeAnimationDurationMs * 0.7)
+      )
+
+  }
+
+  render() {
+    const { index, exitLeft, exitRight, closing, focus } = this.state
+    const { onClose, items } = this.props
+    const { src, caption, credit, alt } = items[index]
+    const total = this.props.items.length
+    const srcs = FigureImage.utils.getResizedSrcs(src, window.innerWidth)
+    return (
+      <div {...styles.wrapper}>
+        <Swipeable
+          onSwipedDown={this.handleSwipeDown}
+          onSwipedLeft={this.handleSwipeLeft}
+          onSwipedRight={this.handleSwipeRight}
+          delta={10}
+          preventDefaultTouchmoveEvent={true}
+          stopPropagation={true}
+        >
+          <NavOverlay
+            handleClickLeft={this.handleClickLeft}
+            handleClickRight={this.handleClickRight}
+            handleClick={this.toggleFocus}
+            onClose={onClose}
+          />
+          <div {...styles.gallery}>
+            <div {...styles.header} style={{ opacity: focus ? 0 : 1 }}>
+              <div {...styles.counter}>
+                <span>{index+1}/{total}</span>
+              </div>
+              <div {...styles.close} onClick={onClose}>
+                <Close size={24} />
+              </div>
+            </div>
+            <div {...styles.body}>
+              <div {...merge(
```

Comment:
Since we're trying to avoid merge at render time, and there's no overlapping style properties as far as I see, would something like this also work?

```
<div
  {...styles.mediaItem}
  {...(exitLeft && styles.exitLeft)}
  {...(exitRight && styles.exitRight)}
  {...(closing && styles.closing)}
>
```

Reply:
Excellent idea, thanks!